tests/test_project_caching.py

```diff
@@ -160,12 +181,228 @@ def test_update_path_cache(tmpdir):
     assert remote_get_url(bar) == "file://" + os.fspath(Path('non-existent') / 'there')
 
 
+def test_update_auto_cache(tmpdir):
+    # Test that 'west update --auto-cache' works and does set up the local
+    # cache correctly.
+
+    # The tree of the auto cache looks like following:
+    # (path cache)
+    # ├── bar
+    # │   └── <hash>
+    # │   └── <hash>.info
+    # └── foo
+    #     └── <hash>
+    # │   └── <hash>.info
```

Comment:
Thanks for adding this and the ones in the previous PR but I feel like I'm still missing the connection with the workspace tree structure (`west update --help` says "in the same relative paths as the workspace being updated")

So could you expand all those a bit? Imagine this is an example in documentation for end users. In the longer term I think they should be in some user-visible documentation but here is good enough to get started (and avoids painful reviews across multiple repos #799)